src/data/list/basic.lean

```diff
@@ -1530,8 +1530,51 @@ theorem prod_hom [monoid β] (l : list α) (f : α → β) [is_monoid_hom f] :
 by { simp only [prod, foldl_map, (is_monoid_hom.map_one f).symm],
   exact l.foldl_hom _ _ _ 1 (is_monoid_hom.map_mul f) }
 
+-- `to_additive` chokes on the next few lemmas, so we do them by hand below
+@[simp]
+lemma prod_take_mul_prod_drop :
+  ∀ (L : list α) (i : ℕ), (L.take i).prod * (L.drop i).prod = L.prod
+| [] i := by simp
+| L 0 := by simp
+| (h :: t) (n+1) := by { dsimp, rw [prod_cons, prod_cons, mul_assoc, prod_take_mul_prod_drop], }
+
+@[simp]
+lemma prod_take_succ :
+  ∀ (L : list α) (i : ℕ) (p), (L.take (i + 1)).prod = (L.take i).prod * L.nth_le i p
+| [] i p := by cases p
+| (h :: t) 0 _ := by simp
+| (h :: t) (n+1) _ := by { dsimp, rw [prod_cons, prod_cons, prod_take_succ, mul_assoc], }
+
+/-- A list with product not one must have positive length. -/
+lemma length_pos_of_prod_ne_one (L : list α) (h : L.prod ≠ 1) : 0 < L.length :=
+by { cases L, { simp at h, cases h, }, { simp, }, }
+
 end monoid
 
+@[simp]
+lemma sum_take_add_sum_drop [add_monoid α] :
+  ∀ (L : list α) (i : ℕ), (L.take i).sum + (L.drop i).sum = L.sum
+| [] i := by simp
+| L 0 := by simp
+| (h :: t) (n+1) := by { dsimp, rw [sum_cons, sum_cons, add_assoc, sum_take_add_sum_drop], }
+
+@[simp]
+lemma sum_take_succ [add_monoid α] :
+  ∀ (L : list α) (i : ℕ) (p), (L.take (i + 1)).sum = (L.take i).sum + L.nth_le i p
+| [] i p := by cases p
+| (h :: t) 0 _ := by simp
+| (h :: t) (n+1) _ := by { dsimp, rw [sum_cons, sum_cons, sum_take_succ, add_assoc], }
+
```

Comment:
I think you should now manually add the `to_additive` attribute. Alternatively, I think you can prove these before the multiplicative version, and then just add the `to_additive` attribute to the multiplicative version as usual, and it will pick up the additive version that is already there.

Reply:
Great, done!

Reply:
@semorrison But it seems you didn't push...

Reply:
Seems I did it sometime later?